src/Analyzers/CSharp/Analyzers/UseParameterNullChecking/CSharpUseParameterNullCheckingDiagnosticAnalyzer.cs

```diff
@@ -154,6 +154,22 @@ protected override void InitializeWorker(AnalysisContext context)
                         additionalLocations: new[] { parameterSyntax.GetLocation() },
                         properties: null));
                 }
+                else
+                {
+                    var descendants = statement.DescendantNodesAndSelf(descendIntoChildren: static c => c is StatementSyntax);
```

Comment:
i don't think you need the DescendantNodes check.  This would only now find returns in lambdas/local functions.